pkg/virt-launcher/virtwrap/util/libvirt_helper.go

```diff
@@ -316,6 +319,85 @@ func StartVirtlog(stopChan chan struct{}, domainName string) {
 			time.Sleep(time.Second)
 		}
 	}()
+
+	go func() {
+		const readSerialFilename = "readserial.py"
+		const readSerialScript = "/etc/qemu/scripts/" + readSerialFilename
+		const QEMUSeaBiosDebugPipeMode uint32 = 0666
+
+		err := syscall.Mkfifo(QEMUSeaBiosDebugPipe, QEMUSeaBiosDebugPipeMode)
+		if err != nil {
+			log.Log.Reason(err).Error(fmt.Sprintf("failed creating a pipe for sea bios debug logs: %s", err.Error()))
+			return
+		}
+
+		err = syscall.Chmod(QEMUSeaBiosDebugPipe, QEMUSeaBiosDebugPipeMode)
+		if err != nil {
+			log.Log.Reason(err).Error(fmt.Sprintf("failed executing chmod on pipe for sea bios debug logs: %s", err.Error()))
+			return
+		}
+
+		for {
+			exitChan := make(chan struct{})
+			readSerialScriptCmd := exec.Command("/bin/python3", readSerialScript, "-nf", QEMUSeaBiosDebugPipe)
+
+			cmdStdout, err := readSerialScriptCmd.StdoutPipe()
+			if err != nil {
+				log.Log.Reason(err).Error(fmt.Sprintf("failed to get stdout pipe for %s. error: %s", readSerialFilename, err.Error()))
+				return
+			}
+
+			err = readSerialScriptCmd.Start()
+			if err != nil {
+				log.Log.Reason(err).Error(fmt.Sprintf("failed to start %s for QEMU SeaBios debug logging", readSerialFilename))
+				panic(err)
+			}
+
+			go func() {
+				var err error
+
+				// It can take a few seconds to the pipe to be created
+				for {
+					_, err = os.Stat(QEMUSeaBiosDebugPipe)
+					if !os.IsNotExist(err) {
```

Comment:
I'm not sure I understand the logic here. What happen if Stat() returns some error which is not IsNotExist  ? It seems like you just continue as usual (or I missed something)

Reply:
Sorry, it's a left-over. I'll remove it. Thanks!